### ISSUE TITLE
csv module examples miss newline='' when opening files

### ISSUE THREAD

**Issue by f009c3de-072f-4e60-bcba-f3cd8a4e455c** [python/cpython#75120]
BPO | [30937](https://bugs.python.org/issue30937)
--- | :---
Nosy | @berkerpeksag
PRs | <li>python/cpython#2730</li><li>python/cpython#4129</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-10-26.08:39:50.022>
created_at = <Date 2017-07-15.14:48:05.444>
labels = ['easy', 'type-bug', '3.7', 'docs']
title = "csv module examples miss newline='' when opening files"
updated_at = <Date 2017-10-26.08:39:50.020>
user = 'https://bugs.python.org/Pavel'
```

bugs.python.org fields:
```python
activity = <Date 2017-10-26.08:39:50.020>
actor = 'berker.peksag'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-10-26.08:39:50.022>
closer = 'berker.peksag'
components = ['Documentation']
creation = <Date 2017-07-15.14:48:05.444>
creator = 'Pavel'
dependencies = []
files = []
hgrepos = []
issue_num = 30937
keywords = ['patch', 'easy']
message_count = 4.0
messages = ['298397', '305035', '305037', '305038']
nosy_count = 3.0
nosy_names = ['docs@python', 'berker.peksag', 'Pavel']
pr_nums = ['2730', '4129']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue30937'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by f009c3de-072f-4e60-bcba-f3cd8a4e455c** [comment]
At the very beginning the csv module documentation (https://docs.python.org/3.7/library/csv.html) advises to open files passing newline='' parameter though three examples don't include it:

Here are the examples:

```pycon
1:
>>> import csv
>>> with open('names.csv') as csvfile:
...     reader = csv.DictReader(csvfile)
...     for row in reader:
...         print(row['first_name'], row['last_name'])
...
Eric Idle
John Cleese

>>> print(row)
OrderedDict([('first_name', 'John'), ('last_name', 'Cleese')])
```


2:
import csv

```py
with open('names.csv', 'w') as csvfile:
    fieldnames = ['first_name', 'last_name']
    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)

    writer.writeheader()
    writer.writerow({'first_name': 'Baked', 'last_name': 'Beans'})
    writer.writerow({'first_name': 'Lovely', 'last_name': 'Spam'})
    writer.writerow({'first_name': 'Wonderful', 'last_name': 'Spam'})
```


3:
with open('example.csv') as csvfile:
    dialect = csv.Sniffer().sniff(csvfile.read(1024))
    csvfile.seek(0)
    reader = csv.reader(csvfile, dialect)
    \# ... process CSV file contents here ...

**Comment by berkerpeksag** [comment]
New changeset 275d2d9c4663a1ea8d1f7c8778567a735b0372c1 by Berker Peksag (Ammar Askar) in branch 'master':
bpo-30937: Make usage of newline='' consistent in csv docs (GH-2730)
https://github.com/python/cpython/commit/275d2d9c4663a1ea8d1f7c8778567a735b0372c1

**Comment by berkerpeksag** [comment]
New changeset 614ea48986f80d361043510ac3945f3dcd666c31 by Berker Peksag (Miss Islington (bot)) in branch '3.6':
bpo-30937: Make usage of newline='' consistent in csv docs (GH-2730)
https://github.com/python/cpython/commit/614ea48986f80d361043510ac3945f3dcd666c31

**Comment by berkerpeksag** [comment]
Good catch, Pavel. Thanks! This is now fixed in 3.6 and 3.7 docs. Thanks for the patch, Ammar.
